### ISSUE TITLE
useSuspenseQuery leaves component suspended if fetch is aborted

### ISSUE THREAD

**Issue by lounsbrough** [apollographql/apollo-client#11270]
### Issue Description

When Apollo aborts a long-running request being run from `useSuspenseQuery`, the component stays suspended instead of throwing to the error boundary.

https://github.com/apollographql/apollo-client/assets/8313853/bb7b1628-5f93-40ba-ac9d-5715aba95140

### Link to Reproduction

https://codesandbox.io/s/clever-rosalind-dhk9sw?file=/src/App.jsx

### Reproduction Steps

- Add a short timeout in the Apollo Client config to make reproducing easier
  - ```js
     defaultOptions: {
       react: {
         suspense: {
           autoDisposeTimeoutMs: 1000
         }
       }
     }
     ``` 
- Run a query with `useSuspenseQuery` that takes longer to return than the `autoDisposeTimeoutMs` settings
- Notice that the `fallback` of the `<Suspense>` component continues to be rendered, whereas the `<ErrorBoundary>` should instead be hit since the request has failed (aborted).

**Comment by jerelmiller** [comment]
Hey @lounsbrough 👋 

This does indeed look like an issue! I do agree that the current behavior is not reasonable, but I'll have to play around with this a bit more in depth as I'm not sure simply rejecting the promise will be enough. That `autoDisposeTimeoutMs` exists to ensure that if a component suspends but never remounts, it gets properly cleaned up. This is why the default value here is so high (30 sec). It is not intended as a means to cancel or abort a request if it reaches a certain threshold. 

That being said, the tricky part about the auto dispose behavior is that even if we reject the promise, auto dispose removes the query ref (our wrapper around the fetch/cache behavior in for suspense components) from the suspense cache, which means that when the component mounts again, it will kick off that query again. This might get you stuck in an infinite request loop until the request is able to complete before that auto dispose timeout.

React's `<Suspense />` component itself won't trigger the error boundary when the promise rejects, which is why simply adding this behavior likely won't resolve the issue (and again could make it worse). `<Suspense />` simply waits for the promise to resolve/reject before deciding whether to try rendering your components again. `useSuspenseQuery` is actually the thing that throws errors for rejected promises.

That's probably a lot more behind-the-scenes info than you care about, but all this to say I'll see if I can figure out a reasonable behavior here. For now, I'd recommend setting `autoDisposeTimeoutMs` to a higher threshold if your requests are taking greater than the default 30 sec to complete.

**Comment by jerelmiller** [comment]
Hey @lounsbrough 👋 

I've got a fix for this in #11274. I've adjusted the `autoDisposeTimeoutMs` to start the timer only after the initial promise settles. 

I've updated your [reproduction](https://codesandbox.io/s/stoic-chaplygin-9c2h78?file=/src/App.jsx) with a release snapshot from #11274 that shows the fix in place. We should (hopefully) be able to get this in the next patch release. Thanks again for reporting this!
